packages/react-components/cypress/utils/types.ts

```diff
@@ -0,0 +1,16 @@
+import React, { JSXElementConstructor } from 'react'
+import type { SinonStub } from 'cypress/types/sinon'
+
+export type PickEventKeys<T extends Record<string, unknown>, TPrefix extends string = 'on'> = keyof T extends infer TKey
+    ? TKey extends `${TPrefix}${string}`
+        ? TKey
+        : never
+    : never
+
+export type PickComponentEventKeys<T extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>> = PickEventKeys<
+    React.ComponentProps<T>
+>
+
+export type ComponentEventStubs<T extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>> = Required<
```

Comment:
@pshoniuk can you add a comment here explaining what this does, it's clear from the code but would be nice to have a little reference here

Reply:
I added a little clarification